### DISCUSSION TITLE
[SUGGESTION] Reuse/adapt sources from other projects

### DISCUSSION THREAD

**Discussion by MarcoRavich** [QAppsSoft/MultiConverter#68] [General]
Hi there, this project aim is very interesting BTW "reinvent the weel" - at least for apps' backend - it's not an efficent approach IOHO.

Since we believe that collaboration between open softwares could push-forward a mutual evolution, we are collecting - using purpose-oriented formatting - these resources in our [HyMPS project](https://github.com/forart/HyMPS#readme) .

You can find a raw (not yet sorted) list of them [Video](https://github.com/forart/HyMPS#-2) / [Tools subsection](https://github.com/forart/HyMPS/blob/main/VideoTools.md#subsections-) that maybe useful for MultiConverter development too.

Hoping that can help - or at least inspire - you, thanks for your (voluntary) work.

**Comment by alex6dj** [comment]
Hello. First of all, thanks for coming here. Second, as English is not my primary language, I apologize in advance for any misunderstanding.

Thank you for looking at my project and thank you very much for your suggestions. This would be my first serious opensource project and of course I'm not a professional developer so everything you can see here I've learned self-taught.

For this project I plan to use several community tools, some of which I have already used and contributed to, in code or finding bugs.

A non-exhaustive list of these would be:
https://github.com/AvaloniaUI/Avalonia/ - GUI
https://github.com/rosenbjerg/FFMpegCore/ - FFmpeg backend
https://github.com/reactiveui/ReactiveUI/
https://github.com/microsoft/vs-streamjsonrpc/ -IPC

There are other projects that I hope to use, but I am still at the beginning of the development of this application. Although it's not going to be a complete rewrite, I plan to reuse parts that I already have from the original implementation.

This application would be to solve a very specific problem that I have in my daily work, although due to its characteristics I know that it can be very useful for other people. I am aware that there are other options to convert, several of which I have already used, but unfortunately they do not cover all my needs. Also this for me is a journey to learn :wink:.

I repeat, thanks for the suggestions, I will take them into account during development.

**Comment by MarcoRavich** [comment]
Thank you for the fast reply.
Don't worry about the english language, wich isn't our primary lang. too (that's why we always try to be clear and simple as possible and, of course, we exploit some translation webservices).

Browsing resources you linked, we deduced that your project will be .NET-centric (so Windows-oriented)...
We - of course - are more in favor of platform-independency for code portability, but we also understand that it's not always practicable (nor, sometimes, attractive) for all devs.

Anyway the **Remote and/or local conversion**-item in the [TODO list](https://github.com/QAppsSoft/MultiConverter#todo) seems to reveal the software's usage kind targeted, therefore we strongly suggest you to consider @master-of-zen's [Av1an](https://github.com/master-of-zen/Av1an#av1an) techniques/approaches that may fit some "scalability" aspects (ex. _Splitting video by scenes for parallel_ - and resumable - _encoding_ using [PySceneDetect](https://pyscenedetect.readthedocs.io/en/latest/)).

It would be also extremely cool to squeeze out all available hardware's computing-power by exploiting @rigaya's [*Enc](https://github.com/rigaya?tab=repositories)s accelerated encoders.

For the **Conversion Presets**-item goal, it might be interesting to adopt @jeanslack's [Videomass presets](https://github.com/jeanslack/Videomass-presets) compatibility or foresee something like that (ex. XML-based presets ?) anyway.

We suggest to take advantage of the detailed reports that @fsinapsi's [AVInaptic2](https://github.com/fsinapsi/avinaptic2) and/or @MediaArea's [MediaInfo](https://github.com/MediaArea/MediaInfo)/[MediaInfoLib](https://github.com/MediaArea/MediaInfoLib) are able to generate for the **Compare video info** item.

It would also be useful - for users who wish - to read the more-technical @fifonik's [FFMetrics](https://github.com/fifonik/FFMetrics) and/or @FranceBB's [VideoTek](https://github.com/FranceBB/VideoTek)/[mpv lua scripts](https://github.com/FranceBB/mpv-scripts) videoscope reports.

Last but not least, thanks to @RxLaboratory's [DuME](https://rxlaboratory.org/tools/dume/), we discovered the (widely adopted) @AcademySoftwareFoundation's [OpenColorIO](https://opencolorio.org/) framework, that we suggest you to consider for color management/manipulation.

Hope that inspires !

**Comment by alex6dj** [comment]
Hi again, thank agains for the suggestions. I love DuME, it almost have all the thing I need.

So this is the app Im already migrating, running right now:
![Captura de pantalla (55)](https://user-images.githubusercontent.com/27690651/217305611-2bd67bb5-8652-4e82-9617-3e78d0256d92.png)

Why, because is build in WPF and is not cross-platform. Yes Im only using windows right now, but in the future this can change. The decision to use Avalonia wasnt deliberate. It is cross-platform, Im already using it in a personal app running in my Windows PC and my Android phone. Same code, same look. It can easely be ported to Linux, Mac, Wasm, or everything you want. Im not a professional programmer, this is only a hobby, and I love it.
